packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -543,6 +541,31 @@ function updateTextContent(vnode: VText | VComment, renderer: RendererAPI) {
     }
 }
 
+function insertFragment(
+    vnode: VFragment,
+    parent: Node,
+    anchor: Node | null,
+    renderer: RendererAPI
+) {
+    if (process.env.NODE_ENV !== 'production') {
+        unlockDomMutation();
+    }
+
+    const children = vnode.children;
+    const nodesToInsert = [];
+    for (let i = 0; i < children.length; i += 1) {
+        const child = children[i];
+        if (!isNull(child)) {
+            nodesToInsert.push(child.elm!);
+        }
+    }
+    nodesToInsert.forEach((node) => renderer.insert(node, parent, anchor));
```

Comment:
Let's use a regular `for` loop here (or the `forEach` from `@lwc/shared`).

Reply:
On second thought, do we even need two separate `for` loops here? We create the array `nodesToInsert` with the first loop, then use the second loop to insert those nodes... So can these two `for` loops be combined to avoid creating the temporary array?